gui/wxpython/lmgr/frame.py

```diff
@@ -426,6 +427,155 @@ def _createPythonShell(self, parent):
         else:
             self.pyshell = None
 
+    def OnNewDisplay(self, event=None):
+        """Create new layer tree and map display window instance"""
+        self.NewDisplay()
+
+    def NewDisplay(self, name=None, show=True):
+        """Create new layer tree structure and associated map display and
+        add it to display notebook tab
+        :param name: name of new map display window
+        :param show: show map display window if True
+        """
+        Debug.msg(1, "GMFrame.NewDisplay(): idx=%d" % self.displayIndex)
+        if not name:
+            name = _("Map Display {number}").format(number=self.displayIndex + 1)
+
+        # make a new page in the bookcontrol for the layer tree (on page 0 of
+        # the notebook)
+        self.pg_panel = wx.Panel(self.notebookLayers, id=wx.ID_ANY, style=wx.EXPAND)
+        self.notebookLayers.AddPage(page=self.pg_panel, text=name, select=True)
+        self.currentPage = self.notebookLayers.GetCurrentPage()
+
+        def CreateNewMapDisplay(layertree):
+            """Callback function which creates a new Map Display window
+            :param layertree: layer tree object
+            :param name: name of new map display window
+            :return: reference to mapdisplay instance
+            """
+            # count map display frame position
+            pos = wx.Point((self.displayIndex + 1) * 25, (self.displayIndex + 1) * 25)
+
+            # create superior Map Display frame
+            mapframe = wx.Frame(layertree,
+                                id=wx.ID_ANY,
+                                pos=pos,
+                                size=globalvar.MAP_WINDOW_SIZE,
+                                style=wx.DEFAULT_FRAME_STYLE,
+                                title=name)
+
+            # create instance of Map Display interface
+            self._gifaceForDisplay = LayerManagerGrassInterfaceForMapDisplay(
+                self._giface, layertree
+            )
+            # create Map Display
+            self.currentPage.mapdisplay = MapDisplay(
```

Comment:
Originally, mapdisplay was attribute of layertree, not notebook page, so I think it should stay that way to not introduce more changes than necessary.
Just return mapdisplay (not `self.currentPage.mapdisplay`) and then pass it to _setupMapDisplay.